agrirouter-api-java-impl/src/main/java/com/dke/data/agrirouter/impl/onboard/OnboardingServiceImpl.java

```diff
@@ -49,15 +54,23 @@ private OnboardingRequest createOnboardingRequestBody(OnboardingParameters param
   private OnboardingResponse onboard(String registrationCode, OnboardingRequest onboardingRequest) {
     this.getNativeLogger()
         .info("BEGIN | Onboarding process. | '{}', '{}'.", registrationCode, onboardingRequest);
+    this.lastError = "";
     Response response =
         RequestFactory.bearerTokenRequest(this.environment.getOnboardUrl(), registrationCode)
             .post(Entity.json(onboardingRequest));
-    this.assertStatusCodeIsCreated(response.getStatus());
-    OnboardingResponse onboardingResponse = response.readEntity(OnboardingResponse.class);
+    try {
+      response.bufferEntity();
+      this.lastError = response.readEntity(String.class);
+      this.assertStatusCodeIsCreated(response.getStatus());
+      this.lastError = "";
```

Comment:
This leads to confusion. Setting the error to reset it later is bad style. Perhaps it is better to put the error - if there is one - into the exception to handle it all at one place.

Reply:
Done here, done with revoke, done with secured